### DISCUSSION TITLE
Is there a function to copy a string to the system clipboard?

### DISCUSSION THREAD

**Discussion by olrtg** [neovim/neovim#27312] [Q&A] [answered]
As the title says.. a function in the Neovim/Lua API to copy a string to your system clipboard. I have done some research but couldn't find anything useful. But here are some things that I've found:

- `vim.fn.copy()` doesn't do that.
- `clipboard` in the help only talks about how Neovim uses some kind of service like `pbcopy` to copy stuff to the system clipboard. I wonder if I should just call those commands (but I wouldn't like to handle OS differences).
- In theory I'm using `unnamedplus` so I can try to put that string in the `"+"` register. But didn't find a way to do that programmatically too.

Would love some help (and correction if I said something not accurate). Thanks in advance!

**Comment by zeertzjq** [comment, marked as answer]
`setreg()`

**Reply by olrtg** [reply]
Oh god... I forgot that there's a lot of abbreviations in the vim API.

Thank you so much! It's working.

**Reply by ghost** [reply]
example for future reference

```
:lua vim.fn.setreg("+", "xxxxxxx")
```
